src/libraries/System.Net.Security/src/System/Net/CertificateValidationPal.Unix.cs

```diff
@@ -88,14 +83,11 @@ internal static partial class CertificateValidationPal
             }
             finally
             {
-                if (remoteContext != null)
+                if (remoteCertificate != IntPtr.Zero)
                 {
-                    if (gotReference)
-                    {
-                        remoteContext.DangerousRelease();
-                    }
-
-                    remoteContext.Dispose();
+                    // Creating X509Certificate will grab refference
+                    // and we need to release it explicitly on failure.
```

Comment:
Not sure the comment makes sense with the code.

```suggestion
                    // Creating X509Certificate will increase the reference count
                    // so we need to release it explicitly on either success or failure.
```

(or whatever)